rbac/management/audit_log/model.py

```diff
@@ -58,3 +62,51 @@ class AuditLog(TenantAwareModel):
     resource_id = models.IntegerField(null=True)
     action = models.CharField(max_length=32, choices=ACTION_CHOICES)
     tenant = models.ForeignKey(Tenant, on_delete=models.SET_NULL, null=True)
+
+    def get_resource_item(self, r_type, request, *args, **kwargs):
+        """Find related information (eg, name, id, etc...) for each resource item."""
+        if r_type == AuditLog.ROLE:
+            if request.data != {}:
+                role_object = get_object_or_404(Role, name=request.data["name"])
+            else:
+                role_object = kwargs["kwargs"]
+            # retrieve role id and name
+            role_object_id = role_object.id
+            role_object_name = "role: " + role_object.name
+            return role_object_id, role_object_name
+
+        elif r_type == AuditLog.GROUP:
+            if request._data is not None:
+                group_object = get_object_or_404(Group, name=request.data["name"])
+            else:
+                group_uuid = kwargs["kwargs"]["uuid"]
+                group_object = get_object_or_404(Group, uuid=group_uuid)
+            group_object_id = group_object.id
+            group_object_name = "group: " + group_object.name
+            return group_object_id, group_object_name
+
+        elif r_type == AuditLog.PERMISSION:
+            # TODO: update for permission related items
+            return None
+
+        elif r_type == "principal":
+            current_user = management.utils.get_principal_from_request(request)
+            principal_object = get_object_or_404(Principal, username=current_user.username)
```

Comment:
tenant info here

Reply:
Thank you so much for the review!! 

Oh! What I was thinking was that if the role is unable to be created correctly like here, such as when there is a duplicate role name within the same tenant (code here:  https://github.com/RedHatInsights/insights-rbac/pull/1103/files#diff-bd923776533ca0c9f2a6a931de1b8c2ccf01c1e37b6b6b8141ee7a2c7d0d507fR212), the audit log will not be created. 
Or is the admin able to view roles/create roles within a different tenant? .....sorry I think I might have gotten myself confused... :sweat_smile: